x-pack/solutions/security/plugins/security_solution/common/detection_engine/prebuilt_rules/diff/extract_rule_data_query.ts

```diff
@@ -81,3 +82,20 @@ export const extractRuleEsqlQuery = (
     language,
   };
 };
+
+/**
+ * Removes the null `alias` field that gets appended from the internal kibana filter util for comparison
+ * Relevant issue: https://github.com/elastic/kibana/issues/202966
+ */
+const normalizeFilterArray = (filters: RuleFilterArray | undefined): RuleFilterArray => {
+  if (filters && filters.length > 0) {
+    return (filters as Filter[]).map((filter) => {
+      if (filter.meta && filter.meta.alias == null) {
+        return { ...filter, meta: { ...filter.meta, alias: undefined } };
+      }
+      return filter;
+    });
+  } else {
+    return [];
+  }
+};
```

Comment:
Don't mind improving readability and making it shorter? On top of that does it make any sense to use loose equality `alias == null`? Loose equality matches `null` and `undefined` in that case. But we wanna replace `null` with `undefined`. When `undefined` is matched the output is the same.

```suggestion
const normalizeFilterArray = (filters: RuleFilterArray | undefined): RuleFilterArray => {
  if (!filters?.length) {
    return [];
  }

  return (filters as Filter[]).map((filter) => ({
    ...filter,
    meta: filter.meta
      ? {
          ...filter.meta,
          alias: filter.meta.alias ?? undefined,
        }
      : undefined,
  }));
};
```